client/dashboard/task-list/tasks/shipping/rates.js

```diff
@@ -0,0 +1,225 @@
+import classnames from 'classnames';
+import { Component, Fragment } from '@wordpress/element';
+import { FormToggle } from '@wordpress/components';
+import { get, xor } from 'lodash';
+import PropTypes from 'prop-types';
+
+/**
+ * WooCommerce dependencies
+ */
+import { Flag } from '@woocommerce/components';
+import { getCurrencyFormatString } from '@woocommerce/currency';
+
+class ShippingRates extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			zoneRates: {},
+			enabledZones: [],
+		};
+
+		this.handleSubmit = this.handleSubmit.bind( this );
+	}
+
+	componentDidUpdate( prevProps ) {
+		const { shippingZones } = this.props;
+
+		if ( prevProps.shippingZones !== shippingZones ) {
+			const zoneRates = {};
+			const enabledZones = [];
+
+			shippingZones.forEach( zone => {
+				const flatRateMethods =
+					zone.methods && zone.methods.length
+						? zone.methods.filter( method => 'flat_rate' === method.method_id )
+						: [];
+				const rate = flatRateMethods.length
+					? flatRateMethods[ 0 ].settings.cost.value
+					: getCurrencyFormatString( 0 );
+				zoneRates[ zone.id ] = rate;
+
+				if ( zone.toggleEnabled && flatRateMethods.length && flatRateMethods[ 0 ].enabled ) {
+					enabledZones.push( zone.id );
+				}
+			} );
+
+			/* eslint-disable react/no-did-update-set-state */
+			this.setState( { zoneRates, enabledZones } );
+			/* eslint-enable react/no-did-update-set-state */
+		}
+	}
+
+	async handleSubmit() {
+		const { enabledZones, zoneRates } = this.state;
+		const { createNotice, shippingZones } = this.props;
+
+		shippingZones.map( zone => {
+			const flatRateMethods = zone.methods
+				? zone.methods.filter( method => 'flat_rate' === method.method_id )
+				: [];
+
+			if ( zone.toggleEnabled && ! enabledZones.includes( zone.id ) ) {
+				// Disable any flat rate methods that exist if toggled off.
+				if ( flatRateMethods.length ) {
+					flatRateMethods.map( method => {
+						apiFetch( {
+							method: 'POST',
+							path: `/wc/v3/shipping/zones/${ zone.id }/methods/${ method.instance_id }`,
+							data: {
+								enabled: false,
+							},
+						} );
+					} );
+				}
+				return;
+			}
+
+			if ( flatRateMethods.length ) {
+				// Update the existing method.
+				apiFetch( {
+					method: 'POST',
+					path: `/wc/v3/shipping/zones/${ zone.id }/methods/${ flatRateMethods[ 0 ].instance_id }`,
+					data: {
+						enabled: true,
+						settings: { cost: zoneRates[ zone.id ] },
+					},
+				} );
+			} else {
+				// Add new method if one doesn't exist.
+				apiFetch( {
+					method: 'POST',
+					path: `/wc/v3/shipping/zones/${ zone.id }/methods`,
+					data: {
+						method_id: 'flat_rate',
+						settings: { cost: zoneRates[ zone.id ] },
+					},
+				} );
+			}
+		} );
+
+		// @todo This is a workaround to force the task to mark as complete.
+		// This should probably be updated to use wc-api so we can fetch shipping methods.
```

Comment:
I see you do include a note about this. Maybe have it at the top of the other file as well?

Reply:
Done!